daemon/tests/happy_path.rs

```diff
@@ -186,3 +222,68 @@ async fn maker_notices_lack_of_taker() {
         next(maker.connected_takers_feed()).await.unwrap()
     );
 }
+
+/// Hide the implementation detail of arriving at the Cfd open state.
+/// Useful when reading tests that should start at this point.
+/// For convenience, returns also OrderId of the opened Cfd.
+async fn start_from_open_cfd_state() -> (Maker, Taker, OrderId) {
+    let heartbeat_interval = Duration::from_secs(60);
+    let maker_listener = tokio::net::TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let mut maker = Maker::start(
+        &MakerConfig::default().with_heartbeat_interval(heartbeat_interval),
+        maker_listener,
+    )
+    .await;
+    let mut taker = Taker::start(
+        &TakerConfig::default().with_heartbeat_timeout(heartbeat_interval * 2),
+        maker.listen_addr,
+        maker.identity_pk,
+    )
+    .await;
+
+    is_next_none(taker.order_feed()).await.unwrap();
+
+    maker.publish_order(dummy_new_order()).await;
+
+    let (_, received) = next_order(maker.order_feed(), taker.order_feed())
+        .await
+        .unwrap();
+
+    taker.mocks.mock_oracle_announcement().await;
+    maker.mocks.mock_oracle_announcement().await;
+
+    taker.take_order(received.clone(), Usd::new(dec!(5))).await;
+    let (_, _) = next_cfd(taker.cfd_feed(), maker.cfd_feed()).await.unwrap();
+
+    maker.mocks.mock_party_params().await;
+    taker.mocks.mock_party_params().await;
+
+    maker.mocks.mock_monitor_oracle_attestation().await;
+    taker.mocks.mock_monitor_oracle_attestation().await;
+
+    maker.mocks.mock_oracle_monitor_attestation().await;
+    taker.mocks.mock_oracle_monitor_attestation().await;
+
+    maker.mocks.mock_monitor_start_monitoring().await;
+    taker.mocks.mock_monitor_start_monitoring().await;
+
+    maker.accept_take_request(received.clone()).await;
+
+    let (taker_cfd, maker_cfd) = next_cfd(taker.cfd_feed(), maker.cfd_feed()).await.unwrap();
+    assert!(matches!(taker_cfd.state, CfdState::ContractSetup { .. }));
+    assert!(matches!(maker_cfd.state, CfdState::ContractSetup { .. }));
```

Comment:
Is this necessary in the setup function?

Reply:
the `next_cfd` call is crucial, otherwise we try to trigger events too early :(
I like describing with the assertions what state we are in - it gives more context to the reader. As this function is moved into a helper, I think it shouldn't decrease readability of the tests.

Reply:
Next week I'll try to come up with a function that just waits until you're in a particular state on the cfd, a one-liner that would be much more readable than this code littered everywhere. I've got some ideas, I'll ping you on Monday :)